apis/field_error.go

```diff
@@ -73,10 +73,90 @@ func (fe *FieldError) ViaField(prefix ...string) *FieldError {
 				strings.Join(append(prefix, oldPath), "."))
 		}
 	}
+
+	newPaths = flattenIndices(newPaths)
```

Comment:
I'm wondering if all of `flattenIndices` can be reduced to the following in the loop above?
```go
		if oldPath == CurrentField || strings.HasPrefix(oldPath, "[") {
			newPaths = append(newPaths, strings.Join(prefix, "."))
			newPaths = newPaths + oldPath
		} else {
			newPaths = append(newPaths,
				strings.Join(append(prefix, oldPath), "."))
		}
```